website/docs/reference/commands/deps.md

```diff
@@ -60,26 +60,34 @@ Update your versions in packages.yml, then run dbt deps
+## Predictable package installs
 
-When you update the package spec and run `dbt deps` again, the `package-lock.yml` and `packages.yml` files update accordingly. 
+Starting in dbt Core v1.7, dbt generates an additional `package-lock.yml` file in the root of your project. This contains the full set of resolved packages, based on the `packages` configuration (in `dependencies.yml` or `packages.yml`). Each subsequent invocation of `dbt deps` will install from the _locked_ set of packages specified in this file. By storing the full set of required packages (with pinned versions) in version-controlled code, it ensures predictable installs in production and consistency across all developers and environments.
 
-There are two flags related to `package-lock.yml`:
-- `dbt deps --lock` &mdash; creates or updates the `package-lock.yml` file but does not install the packages.
-- `dbt deps --upgrade` &mdash; creates or updates the `package-lock.yml` file with the most recent dependencies from `packages.yml`. Also install the packages unless the `--lock` flag is also passed.
+The `package-lock.yml` file includes a `sha1_hash` of the `packages` config. This enables dbt to detect if the `packages` config has updated, and to rerun dependency resolution. If you want to only check for changes to the `packages` config and update the lock file accordingly (without actually installing those packages), you can provide the `--lock` option (that is, `dbt deps --lock`).
+
+It's also possible to force package resolution to rerun, even if the `packages` config hasn't changed, by running `dbt deps --upgrade`. For example, this enables you to always get the latest commits from the `main` branch of an internally maintained `git` package while accepting the risk of unpredictable builds. An alternative to always running `dbt deps --upgrade` in production is to "ignore" the lock file by adding `package-lock.yml` to your project's `.gitignore` file. If you pursue either approach, we strongly recommend adding version pins for third-party packages within your `packages` config.
+
+## Add specific packages
+
+You can use the `dbt deps` command to add or update an existing package configuration &mdash; no need to remember the exact syntax for package configurations. 
+
+For Hub packages (default), which are the easiest to install:
 
-Examples of the `--add-package` flag:
 ```shell
-# add package from hub (--source arg defaults to "hub")
 dbt deps --add-package dbt-labs/dbt_utils@1.0.0
 
-# add package from hub with semantic version range
+# with semantic version range
 dbt deps --add-package dbt-labs/snowplow@">=0.7.0,<0.8.0"
+```
 
+For Git or local packages, use the `--source` option:
+```shell
```

Comment:
improved example chunking for faster scanning

Reply:
I like it. For this line in particular, how about just:
```
For other package types, use the `--source` option:
```